src/vs/editor/contrib/suggest/browser/suggestModel.ts

```diff
@@ -424,23 +425,12 @@ export class SuggestModel implements IDisposable {
 	}
 
 	private _refilterCompletionItems(): void {
-		// Re-filter suggestions. This MUST run async because filtering/scoring
-		// uses the model content AND the cursor position. The latter is NOT
-		// updated when the document has changed (the event which drives this method)
-		// and therefore a little pause (next mirco task) is needed. See:
-		// https://stackoverflow.com/questions/25915634/difference-between-microtask-and-macrotask-within-an-event-loop-context#25933985
-		Promise.resolve().then(() => {
-			if (this._state === State.Idle) {
```

Comment:
@alexdima In past this microtask delay was needed because document change listeners that read the position of the matching editor had a chance of retrieving the old position. 

I believe we briefly touched on this in the past that it isn't needed anymore and my testing shows that this might be true. Please confirm and review

Reply:
Yes, this is safe now since https://github.com/microsoft/vscode/commit/00b43acca3174313700d5e78ec0c61c794841cac